### ISSUE TITLE
Adding support for new SoCs 

### ISSUE THREAD

**Issue by anku-anand** [platformio/platform-shakti#11]
Hi,

Shakti platform has 3 new SoCs. 
1) Pinaka - RV32IMAC - on Artix7 35T board
2) Parashu - RV32IMAC - on Artix7 100T board
3) Vajra - RV64IMACSU - on Artix7 100T board

We have updated  created json files for each SoC.
We also have updated tool chain and openocd.
Kindly let me know how should I submit these changes to platformio/platform-shakti.

Thanks & Regards,
Anand

**Comment by nsathya123** [comment]
@valeros @ivankravets Hi Guys
Please help here.  We have a updated sdk and 3 new board entries.  We also have a custom toolchain. Please help in upgrading the same in pio

**Comment by ivankravets** [comment]
Please provide all the download links for toolchains and SDK.

**Comment by anku-anand** [comment]
Hi,

Following are the links for SDK, tool chain and openocd.

SDK  -- https://gitlab.com/shaktiproject/software/shakti-sdk

Tool chains
SoC       |  toolchain URL
------------|-----------------------
pinaka    | https://gitlab.com/shaktiproject/software/riscv32
parashu | https://gitlab.com/shaktiproject/software/riscv32
vajra      | https://gitlab.com/shaktiproject/software/riscv64

Openocd  - https://gitlab.com/shaktiproject/software/shakti-tools


Thanks & Regards,
Anand

**Comment by anku-anand** [comment]
We do have json files for each of SoC in
https://gitlab.com/shaktiproject/software/platform-shakti/-/tree/develop/boards

svd files in
https://gitlab.com/shaktiproject/software/platform-shakti/-/tree/develop/misc/svd

Please let me know if I can create a branch in this repo and bring in all the changes here.

**Comment by valeros** [comment]
Hi @anku-anand ! I see your toolchains differ from the packages we use for all RISC-V based platforms.
But you only have toolchains for Linux, what to do with MacOS and Windows?

**Comment by anku-anand** [comment]
Hi,
@valeros ,
We will use the tool chain provided by PIO.  Just need the openocd from https://gitlab.com/shaktiproject/software/shakti-tools

Thanks

**Comment by valeros** [comment]
@anku-anand Where can I download your OpenOCD for Windows and Mac?

**Comment by bil-ash** [comment]
@anku-anand I would also like to use one of the three boards(for Swadeshi Processor Challenge) from PIO on Windows. So please provide toolchains and OpenOCD for windows. It would be fine even if only OpenOCD is provided, I shall use the toolchains on Ubuntu on [WSL](https://docs.microsoft.com/en-us/windows/wsl/about)

**Comment by anku-anand** [comment]
@valeros ,
Hi,
We have changed ftdi.cfg so that we can use openocd provided by PIO.
SDK -- https://gitlab.com/shaktiproject/software/shakti-sdk
Board files are in https://gitlab.com/shaktiproject/software/platform-shakti

Thanks & Regards,
Anand

**Comment by valeros** [comment]
Hi @anku-anand ! Looks like the https://gitlab.com/shaktiproject/software/platform-shakti repository is private.

**Comment by anku-anand** [comment]
Hi,
Sorry, did not notice that, I will get that updated.
Thanks & Regards,
Anand

**Comment by anku-anand** [comment]
@valeros ,
Hi,
Repo  https://gitlab.com/shaktiproject/software/platform-shakti  is now public.

Thanks & Regards,
Anand

**Comment by valeros** [comment]
Thanks, I've updated the SDK package and synchronized boards with your repository on Gitlab. Please retest with the upstream version of the development platform:
```ini
[env:upstream_develop]
platform = https://github.com/platformio/platform-shakti.git
framework = shakti-sdk
board = ...
```

**Comment by anku-anand** [comment]
Hi,

Thanks a lot.

I ran a simple test on one of the SoC. It is working fine. We will test
with other SoCs and update tomorrow.

Thanks & Regards,
Anand

**Comment by anku-anand** [comment]
@valeros ,
Hi,
Tested with [env:upstream_develop]
It is working fine.

Thanks & Regards,
Anand

**Comment by bil-ash** [comment]
@valeros 
Before closing this issue please ensure that openocd works fine from windows on the three new boards. There is already an [issue](https://github.com/platformio/platform-shakti/issues/10) regarding this (for generic Arty 35t)
Regards,
A windows user

**Comment by valeros** [comment]
@bil-ash Unfortunately, I don't have any of that boards so I cannot test the uploading.

**Comment by bil-ash** [comment]
@anku-anand Sir,please do check if openocd works for Vajra on windows. I am representing one of the teams of Swadeshi Processor Challenge.

**Comment by anku-anand** [comment]
@valeros 
Hi,
This issue corresponds to adding support for the new SoCs in PIO. 

@bil-ash, All our dev environments are Linux based, support in Linux is extensively tested.  We will check openocd in Windows and update you by early next week. Please  mail me anku.anand@gmail.com. 

Thanks & Regards,
Anand

**Comment by bil-ash** [comment]
> @bil-ash, All our dev environments are Linux based, support in Linux is extensively tested. We will check openocd in Windows and update you by early next week. Please mail me [anku.anand@gmail.com](mailto:anku.anand@gmail.com).
> 
> Thanks & Regards,
> Anand

Ok,thanks. I shall mail you next week to know the results.

**Comment by anku-anand** [comment]
@valeros
Hi,

Can you please merge the changes.

Thanks & Regards,
Anand

**Comment by valeros** [comment]
Hi @anku-anand , what do you mean by "merge the changes" ? Do you want me to update the Shakti SDK?

**Comment by bil-ash** [comment]
He most probably meant moving the latest changes from develop to master branch and making a new release with support for the 3 new boards.

**Comment by anku-anand** [comment]
Hi,
@valeros, Please update Shakti SDK.

Thanks & Regards,
Anand

**Comment by anku-anand** [comment]
Hi,
@valeros, Please update Shakti SDK.

Thanks & Regards,
Anand

**Comment by valeros** [comment]
The Shakti SDK has been updated, please let me know if we are ready for a new release.

**Comment by anku-anand** [comment]
Hi,
@veleros, yes we are ready for new release.

Thanks & Regards,
Anand

**Comment by valeros** [comment]
Resolved in [v1.1.0](https://github.com/platformio/platform-shakti/releases/tag/v1.1.0).
